assembly/src/assembler/context.rs

```diff
@@ -250,11 +250,19 @@ impl AssemblyContext {
     // HELPER METHODS
     // --------------------------------------------------------------------------------------------
 
-    /// Returns the context of the procedure currently being complied, or None if module or
+    /// Returns the context of the procedure currently being compiled, or None if module or
     /// procedure stacks are empty.
     fn current_proc_context(&self) -> Option<&ProcedureContext> {
         self.module_stack.last().and_then(|m| m.proc_stack.last())
     }
+
+    /// Returns the name of the procedure currently being compilied, or None if module or
+    /// procedure stacks are empty.
+    pub(crate) fn current_context_name(&self) -> &str {
+        self.current_proc_context()
+            .map(|p| p.name().as_ref())
+            .unwrap_or(ProcedureName::MAIN_PROC_NAME)
+    }
```

Comment:
I forget the exact logic here - but shouldn't we haven an `expect` here instead of `unwrap_or`? Under which circumstances would we get an empty module or procedure stack here?

Reply:
We will have an empty proc name if we are inside of the main block. The `current_proc_context` will fetch the top context from the call stack

Reply:
Isn't main block always initialized with `#main` proc? (i.e., see [here](https://github.com/0xPolygonMiden/miden-vm/blob/main/assembly/src/assembler/context.rs#L290)).

Reply:
We don't always have a context, so we just fallback to the main proc if absent

Reply:
Under what conditions do we end up not having a context here? I'm not sure I can come up with an example of an executable program which doesn't have a context (but I also didn't think about it too much).

Reply:
Not an executable, but libraries compilation is one example:

https://github.com/0xPolygonMiden/miden-vm/blob/159a04f3d4648f77d664642409e521438378bd6c/assembly/src/assembler/context.rs#L297-L307

It would be an edge case, but as we provide debug data bundled into libraries, we could eventually touch that edge case.

Reply:
Done with expect!